modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentResourceImpl.java

```diff
@@ -407,6 +400,29 @@ private Document _addDocument(
 					documentOptional, groupId)));
 	}
 
+	private UnsafeConsumer<BooleanQuery, Exception>
+		_createDocumentPageUnsafeConsumer(Long siteId, Boolean flatten) {
```

Comment:
Not sure about the name of the method. I think it would be better to include BooleanQuery to be more precise about what you're creating. Maybe _createDocumentsPageBooleanQueryUnsafeConsumer. It might be longer but it matches better with the parameter name used by _getDocumentsPage and SearchUtil.search